### PR TITLE
Follow-up: Copy everything, not just one dll

### PR DESCRIPTION
**Pull request by MikaelMayer**

### CODE REVIEW

File: src/language/dafnyInstallation.ts [hunk 1]
```diff
@@ -120,30 +136,41 @@ export async function getOrComputeLanguageServerRuntimePath(context: ExtensionCo
...
+    // Copy all the files from installationDir to vscodeDir
+    const files = await fs.promises.readdir(installationDir);
+    for(const file of files) {
+      if(!(file.endsWith(dll)
+      || file.endsWith(runtimeconfigjson)
+      || file.endsWith(depsjson)
+      || file.endsWith('.pdb')
+      || file === 'z3')) {
+        continue;
+      }
```
Comment (camrein) [inline src/language/dafnyInstallation.ts:160]: nit: I suggest extracting a method here or at least indent the different cases.

File: src/language/dafnyInstallation.ts [hunk 2]
```diff
@@ -120,30 +136,41 @@ export async function getOrComputeLanguageServerRuntimePath(context: ExtensionCo
...
+      // If it's a directory, we use the function above
+      if((await fs.promises.stat(path.join(installationDir, file)))
+        .isDirectory()) {
```
Comment (camrein) [inline src/language/dafnyInstallation.ts:163]: nit: In my opinion, it looks cleaner if the whole condition is on a single line.
```suggestion
      if((await fs.promises.stat(path.join(installationDir, file))).isDirectory()) {
```

File: src/language/dafnyInstallation.ts [hunk 3]
```diff
@@ -120,30 +136,41 @@ export async function getOrComputeLanguageServerRuntimePath(context: ExtensionCo
...
+        await copyDir(path.join(installationDir, file),
+          path.join(vscodeDir, file));
```
Comment (camrein) [inline src/language/dafnyInstallation.ts:165]: nit
```suggestion
        await copyDir(path.join(installationDir, file), path.join(vscodeDir, file));
```

Comment (camrein) [reply src/language/dafnyInstallation.ts:165]: Alternatively, if you prefer multiline, I like to place all arguments on a separate line:
```suggestion
        await copyDir(
          path.join(installationDir, file),
          path.join(vscodeDir, file)
        );
```
Similar to having a nested block of code.

File: src/language/dafnyInstallation.ts [hunk 4]
```diff
@@ -120,30 +136,41 @@ export async function getOrComputeLanguageServerRuntimePath(context: ExtensionCo
...
-    const newRunJson = path.join(installationDir, `${dlsName}-vscode.runtimeconfig.json`);
-    await fs.promises.copyFile(oldRunJson, newRunJson);
-    const vscodedllpath = path.join(installationDir, 'vscode.dll');
-    if(!await existsAsync(vscodedllpath)) {
-      await fs.promises.mkdir(vscodedllpath);
-    }
-    const oldCore = path.join(installationDir, `${dcName}${dll}`);
-    const newCore = path.join(vscodedllpath, `${dcName}${dll}`);
-    await fs.promises.copyFile(oldCore, newCore);
+    const newPath = path.join(vscodeDir, `${dlsName}.dll`);
```
Comment (camrein) [inline src/language/dafnyInstallation.ts:173]: nit: I have the feeling that his comment is relevant for the whole copy procedure and not only this particular dll.

File: src/language/dafnyInstallation.ts [hunk 5]
```diff
@@ -97,6 +97,22 @@ function isNullOrEmpty(s: string | null): boolean {
   return s === '' || s == null;
 }
 
+async function copyDir(src: string, dest: string) {
```
Comment (camrein) [inline src/language/dafnyInstallation.ts:100]: nice!

**Comment by MikaelMayer** [comment]
I'm puzzled by a new problem. If I have two VSCode windows open, it will now try to copy the dlls as well, but it won't be able because the dlls are locked by the other version.

**Pull request merged**